src/client/Client.cc

```diff
@@ -5133,7 +5134,7 @@ void Client::_try_to_trim_inode(Inode *in, bool sched_inval)
     --ref;
   }
 
-  if (ref > 0 && in->ll_ref > 0 && sched_inval) {
+  if (ref > 0  && sched_inval) {
     auto q = in->dentries.begin();
     while (q != in->dentries.end()) {
       Dentry *dn = *q;
```

Comment:
````
  if (ref > 0 ) {
    auto q = in->dentries.begin();
    while (q != in->dentries.end()) {
      Dentry *dn = *q;
      ++q;
      if (sched_inval &&  in->ll_ref > 0) {
      // FIXME: we play lots of unlink/link tricks when handling MDS replies,
      //        so in->dentries doesn't always reflect the state of kernel's dcache.
      _schedule_invalidate_dentry_callback(dn, true);
      }
      unlink(dn, true, true);
    }
  }

````

Reply:
thanks,  I was though the ll_ref > 0 is checked in _schedule_invalidate_dentry_callback